File: services/account-service/models/cuenta.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type TipoCuenta string

const (
	TipoCuentaMonetaria TipoCuenta = "MONETARIA"
	TipoCuentaAhorro    TipoCuenta = "AHORRO"
)

type EstadoCuenta string

const (
	EstadoCuentaActiva    EstadoCuenta = "ACTIVA"
	EstadoCuentaInactiva  EstadoCuenta = "INACTIVA"
	EstadoCuentaBloqueada EstadoCuenta = "BLOQUEADA"
	EstadoCuentaCerrada   EstadoCuenta = "CERRADA"
)

type Cuenta struct {
	IDCuenta           uuid.UUID    `json:"idCuenta"`
	IDCliente          uuid.UUID    `json:"idCliente"`
	NumeroCuenta       string       `json:"numeroCuenta"`
	TipoCuenta         TipoCuenta   `json:"tipoCuenta"`
	SaldoCentavos      int64        `json:"saldoCentavos"`
	Moneda             string       `json:"moneda"`
	Estado             EstadoCuenta `json:"estado"`
	UltimaActividad    *time.Time   `json:"ultimaActividad,omitempty"`
	FechaCreacion      time.Time    `json:"fechaCreacion"`
	FechaActualizacion time.Time    `json:"fechaActualizacion"`
	Version            int64        `json:"version"`
}

File: frontend/bank-usac-web/src/features/accounts/types/cuenta.ts
export type EstadoCuenta='ACTIVA'|'INACTIVA'|'BLOQUEADA'|'CERRADA';
export interface Cuenta{ idCuenta:string;idCliente:string;numeroCuenta:string;tipoCuenta:'MONETARIA'|'AHORRO';saldoCentavos:number;moneda:string;estado:EstadoCuenta;ultimaActividad?:string }
export interface MovimientoCuenta{idMovimiento:string;tipoMovimiento:'DEBITO'|'CREDITO'|'COMPENSACION';montoCentavos:number;saldoNuevoCentavos:number;descripcion?:string;fechaCreacion:string}
export interface OperacionAceptada{operationId:string;correlationId:string;status:string;statusUrl:string}
